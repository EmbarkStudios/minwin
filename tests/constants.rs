mod utils;
use utils::*;

test!(
    regular,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Graphics.OpenGL.GLU_BEGIN",
        "Windows.Win32.Graphics.OpenGL.GLU_CCW",
        "Windows.Win32.Graphics.OpenGL.GL_ADD",
        "Windows.Win32.Graphics.OpenGL.GL_ALL_ATTRIB_BITS",
    ]
);

// Ensures we can generate string constants
test!(
    strings,
    [BindConfig::Minwin(Default::default()), BindConfig::Bindgen,],
    [
        // utf-16
        "Windows.Win32.Devices.Enumeration.Pnp.ADDRESS_FAMILY_VALUE_NAME",
        // ansi
        "Windows.Win32.Media.Multimedia.JOY_CONFIGCHANGED_MSGSTRING",
    ]
);
