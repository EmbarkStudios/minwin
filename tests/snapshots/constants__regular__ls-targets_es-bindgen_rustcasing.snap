---
source: tests/constants.rs
expression: bo.bindings
---
#![allow(non_snake_case, non_upper_case_globals, non_camel_case_types)]
pub const GlAdd: u32 = 260;
pub const GlAllAttribBits: u32 = 1048575;
pub const GluBegin: u32 = 100100;
pub const GluCcw: u32 = 100121;

